packages/react-native/React/CxxBridge/JSCExecutorFactory.h

```diff
@@ -16,12 +16,23 @@ class JSCExecutorFactory : public JSExecutorFactory {
   explicit JSCExecutorFactory(JSIExecutor::RuntimeInstaller runtimeInstaller)
       : runtimeInstaller_(std::move(runtimeInstaller)) {}
 
+  void setEnableDebugger(bool enableDebugger);
+
+  void setDebuggerName(const std::string &debuggerName);
+
   std::unique_ptr<JSExecutor> createJSExecutor(
       std::shared_ptr<ExecutorDelegate> delegate,
       std::shared_ptr<MessageQueueThread> jsQueue) override;
 
  private:
   JSIExecutor::RuntimeInstaller runtimeInstaller_;
+
+#if DEBUG
+  bool enableDebugger_ = true;
+#else
+  bool enableDebugger_ = false;
+#endif
```

Comment:
Oversight caught internally: Initially this was always `true` so release builds would have the debugger enabled. I changed it so the default is true only for debug. This way, one could still enable debugging for a ship/release build if they wished.

I considered copying what Hermes does and putting all debugging code behind a custom preprocessor variable (https://github.com/facebook/react-native/pull/37723), but that seemed overkill and would prevent us from enable debugging on ship. 
For reference, equivalent PRs in React Native macOS:
https://github.com/microsoft/react-native-macos/pull/1976
https://github.com/microsoft/react-native-macos/pull/1977
https://github.com/microsoft/react-native-macos/pull/1978
https://github.com/microsoft/react-native-macos/pull/1979